src/main/java/games/strategy/engine/config/PropertyFileReader.java

```diff
@@ -0,0 +1,56 @@
+package games.strategy.engine.config;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Properties;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+
+/**
+ * Reads property values from the game engine configuration file.
+ * 
+ * @see PropertyReader for a complete listing of property keys
+ */
+public class PropertyFileReader implements PropertyReader {
+
+  private final File propertyFile;
+
+  PropertyFileReader(String propertyFile) {
+    this(new File(propertyFile));
+  }
+
+  @VisibleForTesting
+  PropertyFileReader(final File propertyFile) {
+    this.propertyFile = propertyFile;
+  }
+
+
+  protected String readProperty(final String propertyKey) {
+    try (FileInputStream inputStream = new FileInputStream(propertyFile)) {
+      final Properties props = new Properties();
+      props.load(inputStream);
+
+      if (!props.containsKey(propertyKey)) {
+        throw new PropertyNotFoundException(propertyKey, propertyFile.getAbsolutePath());
+      } else {
+        return props.getProperty(propertyKey).trim();
+      }
+    } catch (final FileNotFoundException e) {
+      throw Throwables.propagate(e);
+    } catch (final IOException e) {
+      throw new IllegalStateException("Failed to read propertyFile: " + propertyFile.getAbsolutePath(), e);
+    }
+  }
+
+  static  class PropertyNotFoundException extends IllegalStateException {
```

Comment:
Re: `public`, IMO use minimum scope. We can increase the scope if we need to if it comes up.